source/includes/dev/_microservice_dev_guide.md

```diff
@@ -8,10 +8,280 @@ Control Tower, which is mentioned throughout these docs, will be replaced soon w
+
+While it's rare, tests ran by Jenkins at this stage may also fail, preventing your deployment. In these cases, refer to the Jenkins build log for details, which most of the times can be reproduced locally [running your tests using Docker](#running-the-tests69). If your Jenkins log mentions issues related with disk capacity or network address assignment problems, please reach out to someone with access to the Jenkins VMs and ask for a [docker system prune](https://docs.docker.com/engine/reference/commandline/system_prune/).
+
+## Testing your deployed code
+
+With your code live on one of the clusters, you should now proceed to testing it. The type of tests you should run vary greatly with the nature of the changes you did, so common sense and industry best practices apply here:
+
+- The bigger the change, the broader the testing should be.
+- Test your changes with different types of users, `applications`, payloads, etc.
+- Try to break your code - send unexpected input, try to access resources you should not have access to, etc. More important than doing what it should is not doing what it shouldn't .
+- If you can, ask for help - testing can be seen as an exercise in creativity, and having someone's assistance will help think outside the box.
+- If you find a bug, fix it, and test everything again, not only what you just fixed.
+- If a test is "simple", write it as a code test, which is reproducible. Save manual testing for the complex scenarios.
+- Test the assumptions you used for behavior of other microservices - E2E testing mocks other microservices, so this may be the first time your code is running alongside real instances of other microservices.
+- Clean up after your tests - if you created a bunch of test data, do your best to delete it once you're done. This is particularly important if you are testing something in the production environment, as that test data may be visible to real world users. Cleaning up in staging is also highly recommended.
+
+If you are implementing a new endpoint and it's mission critical to the RW API or one of the applications it powers, you may want to add an [AWS Cloudwatch Canary](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_Create.html) to ensure that any issue affecting its availability is detected and reported. The best way to create a new Canary is by copying an existing one and modifying the source code to point to your endpoint. Typically, Canaries are only created for the production environment, and should be configured to run once per hour. Don't forget to set up an alarm to go with your Canary, otherwise a notification will not be issued if a failure is detected.
```

Comment:
Please add a reference here to the section in the developer docs about smoke tests: https://resource-watch.github.io/doc-api/developer.html#api-smoke-tests